src/storage/app-state/slices/snackbarSlice.ts

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { SnackbarMessage } from '@/components/SnackbarManager'
+import i18n from '@/i18n'
+import { syncAddressesData } from '@/storage/app-state/slices/addressesSlice'
+import { contactStoredInPersistentStorage } from '@/storage/app-state/slices/contactsSlice'
+import {
+  apiClientInitFailed,
+  apiClientInitSucceeded,
+  customNetworkSettingsSaved
+} from '@/storage/app-state/slices/networkSlice'
+
+const sliceName = 'snackbarSlice'
+
+interface SnackbarSliceState {
+  messages: SnackbarMessage[]
+  offlineMessageWasVisibleOnce: boolean
+}
+
+const initialState: SnackbarSliceState = {
+  messages: [],
+  offlineMessageWasVisibleOnce: false
+}
+
+const displayError = (state: SnackbarSliceState, action: PayloadAction<string>) =>
+  displayMessageImmediately(state, { text: action.payload, type: 'alert', duration: 5000 })
+
+const snackbarSlice = createSlice({
+  name: sliceName,
+  initialState,
+  reducers: {
+    snackbarDisplayTimeExpired: (state) => {
+      if (state.messages.length > 0) state.messages.shift()
+    },
+    passwordValidationFailed: (state) =>
+      displayMessageImmediately(state, { text: i18n.t('Invalid password'), type: 'alert' }),
+    copiedToClipboard: (state) => displayMessageImmediately(state, { text: i18n.t('Copied to clipboard!') }),
+    transactionBuildFailed: displayError,
+    transactionSendFailed: displayError,
+    contactStorageFailed: displayError,
+    walletCreationFailed: displayError,
+    transactionsSendSucceeded: (state, action: PayloadAction<number>) => {
+      const nbOfTransactionsSent = action.payload
+
+      displayMessageImmediately(state, {
+        text: nbOfTransactionsSent > 1 ? i18n.t('Transactions sent!') : i18n.t('Transaction sent!'),
+        type: 'success'
+      })
+    }
+  },
+  extraReducers(builder) {
+    builder
+      .addCase(syncAddressesData.rejected, (state, action) => {
+        const message = action.payload
+
+        if (message) queueMessage(state, message)
+      })
+      .addCase(apiClientInitSucceeded, (state, action) => {
```

Comment:
Linked to previous comment: having side effects like this one scattered around the slices makes it hard to understand what's going on IMO.
What about dispatching a `snackbarMessageShown` action where the `apiClientInitSucceeded` fn is executed (in the network slice)?

Reply:
I understand why you would see this as a side effect, but this is truly where Redux shines! One action may trigger several reducers. There's no one-to-one relationship between action and reducer.

> A given action could be handled by all, some, or none of them. This keeps components decoupled from the actual data changes, as one action may affect different parts of the state tree, and there is no need for the component to be aware of this.
https://redux.js.org/faq/actions#is-there-always-a-one-to-one-mapping-between-reducers-and-actions


> This is only true in trivial examples, but it is extremely limiting in real applications. Beginners exposed to this pattern couple reducers and action creators, and fail to realize they're meant to be many-to-many and decoupled.
>
> Many reducers may handle one action. One reducer may handle many actions. Putting them together negates many benefits of how Flux and Redux application scale. This leads to code bloat and unnecessary coupling. You lose the flexibility of reacting to the same action from different places, and your action creators start to act like “setters”, coupled to a specific state shape, thus coupling the components to it as well.
https://github.com/Pitzcarraldo/reduxible/issues/8

Reply:
Of course! I simplify think that we should be careful about creating too many slices in the long run. It's sometimes difficult to know where to declare which action (cf. previous comment).